eth/client.go

```diff
@@ -162,212 +157,32 @@ func NewClient(accountAddr ethcommon.Address, keystoreDir string, eth *ethclient
 	if err != nil {
 		return nil, err
 	}
+	backend.SetMaxGasPrice(maxGasPrice)
 
 	am, err := NewAccountManager(accountAddr, keystoreDir, signer)
 	if err != nil {
 		return nil, err
 	}
 
+	if err := am.Unlock(password); err != nil {
+		return nil, err
+	}
+
+	// Setting gasLimit and gasPrice to zero, nil allows it to be overwritten by eth.Backend
+	opts, err := am.CreateTransactOpts(0, nil)
+	if err != nil {
+		return nil, err
+	}
+
 	return &client{
 		accountManager: am,
 		backend:        backend,
 		controllerAddr: controllerAddr,
 		txTimeout:      txTimeout,
+		defaultTxOpts:  opts,
 	}, nil
 }
 
-func (c *client) Setup(password string, gasLimit uint64, gasPrice *big.Int) error {
-	err := c.accountManager.Unlock(password)
-	if err != nil {
-		return err
-	}
-
-	return c.SetGasInfo(gasLimit, gasPrice)
-}
-
-func (c *client) SetGasInfo(gasLimit uint64, gasPrice *big.Int) error {
-	opts, err := c.accountManager.CreateTransactOpts(gasLimit, gasPrice)
-	if err != nil {
-		return err
-	}
-
-	if err := c.setContracts(opts); err != nil {
-		return err
-	} else {
-		c.gasLimit = gasLimit
-		c.gasPrice = gasPrice
-		return nil
-	}
-}
-
-func (c *client) GetGasInfo() (gasLimit uint64, gasPrice *big.Int) {
-	return c.gasLimit, c.gasPrice
-}
-
-func (c *client) setContracts(opts *bind.TransactOpts) error {
```

Comment:
Why is this being removed? This will cause breakage because the contract sessions used by the client will all be nil.